src/util/CalibrationUtils.h

```diff
@@ -0,0 +1,107 @@
+#pragma once
+
+namespace daisy
+{
+/** @brief Helper class for calibrating an input to 1V/oct response 
+ *  @author shensley
+ * 
+ *  This provides a scale and offset value for converting
+ *  incoming CV into MIDI note numbers accurately for musical pitch tracking.
+ * 
+ *  To use, record both the 1V and 3V values using the specified functions
+ *  once calibration is complete you can use the GetData function to retrieve
+ *  the calibration values.
+ */
+class VoctCalibration
+{
+  public:
+    VoctCalibration()
+    : c1_(0.f),
+      c3_(0.f),
+      scale_(0.f),
+      offset_(0.f),
+      c1_read_(false),
+      cal_(false)
+    {
+    }
+
+    ~VoctCalibration() {}
+
+    enum class Status
+    {
+        OK,
+        ERR_INVALID_RANGE,
+        ERR_CAL_OUT_OF_ORDER,
+        ERR_CAL_NOT_COMPLETE,
+    };
+
+    /** Record the input value corresponding to 1 Volt */
+    Status Record1V(float val)
+    {
+        c1_      = val;
+        c1_read_ = true;
+        return Status::OK;
+    }
+
+    /** Record the input value corresponding to 3 Volts,
+     *  and calculate the calibration data.
+     */
+    Status Record3V(float val)
```

Comment:
Why not combine the two `Record` functions into one, removing the need for the additional state variables `c1_read_` and `c1_`

Reply:
Do you mean something like:
`Status Record(float val)` that reads the relevant voltage for the current state of the calibrator? I hadn't really thought of doing that, but I like it. I think initially I thought that might be confusing to use because you wouldn't necessarily know what to send it, but that can easily be in the documentation.

Reply:
No, actually I meant to make it a single step, like this:
```cpp
Status Record(float val1V, float val3V);
```
How these values are aquired should be up to the application, I think.

Reply:
Hmm, I can see benefits to both. I'll make your suggested change and see what it looks like on the application side.

Reply:
yeah, the state machine adds a little  bit of clutter to the user app, but in its most limited form its still a lot less than what was built into the class